### DISCUSSION TITLE
Tips for vscode users (hide _templ.go/text files)

### DISCUSSION THREAD

**Discussion by leandergangso** [a-h/templ#520] [Show and tell]
It can be a bit much to have `.templ`, `_templ.go` and `_templ.txt` files in your vscode sidebar.
A quick and nice fix for this is to create a workspace specific settings.json, inside the `.vscode` dir.
Her your can put the following to hide files from your sidebar:
```json
{
    "files.exclude": {
        "**/*_templ.go": true,
        "**/*_templ.txt": true,
    }
}
```
If you want to see the files again, just set the value(s) to `false`.

**Comment by duxnp** [comment]
I think excluding files is a good idea in this case since most likely you would never need to open these generated files. In case you did, here are a couple of other ideas.

There is an extension called Toggle Excluded Files to easily unhide the excluded files, but of course it would be yet another extension to load.

Another potential solution is to use the file nesting feature.

```json
{
    "explorer.fileNesting.patterns": {
        "*.templ": "${capture}_templ.go, ${capture}_templ.txt"
    }
}
```

In the explorer sidebar, the templ file would sort of appear as though it were a folder with the other files contained within. The problem with that is it can be a bit visually confusing if you have other files within the same directory that don't match any of the nesting patterns.

**Reply by sirgallifrey** [reply]
Wow, file nesting is a great tip!
For people like me who didn't even knew this existed in VScode, you also need to enable it:
`"explorer.fileNesting.enabled": true`

**Reply by duxnp** [reply]
Whoops! Nice catch on enabling it. I'm not sure how I forgot that.

**Comment by valentimarco** [comment]
i saw on internet [this post](https://stackoverflow.com/questions/64628565/how-to-update-users-settings-json-in-vscode-with-extension-packs) but my exp in vscode plugin is 0... Is really possbile?
